src/Featurizers/Components/VectorNormsEstimator.h

```diff
@@ -0,0 +1,167 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "../../Traits.h"
+#include "NormUpdaters.h"
+#include "../Structs.h"
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+static constexpr char const * const         VectorNormsEstimatorName("VectorNormsEstimator");
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         VectorNormsAnnotationData
+///  \brief         An annotation class which contains the norms of a matrix
+///
+template <typename UpdaterType>
+class VectorNormsAnnotationData {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    std::vector<std::double_t> const Norms;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    VectorNormsAnnotationData(std::vector<std::double_t> norms);
+    ~VectorNormsAnnotationData(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(VectorNormsAnnotationData);
+};
+
+namespace Details {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         VectorNormsTrainingOnlyPolicy
+///  \brief         `VectorNormsEstimator` implementation details.
+///
+template <typename T, typename UpdaterType>

```

Comment:
>T [](start = 19, length = 1)

Rename this to IteratorRangeT to make it clear what is expected. Also, static_assert that this type is a tuple or pair.